components/Contest.tsx

```diff
@@ -11,19 +11,27 @@ interface ContestProps {
   index: number
 }
 
-export default function Contest({ contest }: ContestProps) {
+export default function Contest({ contest, index }: ContestProps) {
   return (
-    <div className='max-w-sm rounded overflow-hidden shadow-lg '>
-      <img className='w-full' src='/images/contest.jpeg' alt='' />
-
-      <div className='px-6 py-4 text-center'>
-        <div className='font-bold text-xl mb-2'>{contest.name}</div>
-        <p className='text-gray-700 text-base'>{contest.description}</p>
-        <p className='text-gray-700 text-base'>
-          <a>End date: </a>
-          {contest.date_end.slice(0, 10)}
-        </p>
-        <p>{contest.status}</p>
+    <div
+      key={index}
```

Comment:
Contest id is better than using index for key mappings. It will avoid re-rendering previous contests if the request runs again.